jgrapht-core/src/main/java/org/jgrapht/alg/tour/ParberryKnightTour.java

```diff
@@ -0,0 +1,412 @@
+         #****************E**|G******************#
+         #-------------------|-------------------#
+         #******************E|**G****************#
+         #*******************|H******************#
+         #*****************H*|*******************#
+         #******TOUR 3*******|******TOUR 4*******#
+         #*******************|*******************#
+         #*******************|*******************#
+         #*******************|*******************#
+         #*******************|*******************#
+         #########################################
+         */
+
+        /*
+         Relation between nodes in structured array and endpoints of the edges to be deleted/added.
+         Note that you don't know the direction of the edges A, B, C, D, so you have to check both options.
+
+         #########################################
+         #**0***************2|**0***************2#
+         #1******************|1******************#
+         #*****************3*|*****************3*#
+         #******TOUR 1*******|******TOUR 2*******#
+         #*******************|*******************#
+         #*4*****************|*4*****************#
+         #******************6|******************6#
+         #5***************7**|5***************7**#
+         #-------------------|-------------------#
+         #**0***************2|**0***************2#
+         #1***************3**|1******************#
+         #*******************|*****************3*#
+         #******TOUR 3*******|******TOUR 4*******#
+         #*******************|*******************#
+         #*4*****************|*4*****************#
+         #******************6|******************6#
+         #5***************7**|5***************7**#
+         #########################################
+         _________________________________
+
+         A.start = tour1.forStructured[6];
+         A.end = tour1.forStructured[7];
+
+         or
+
+         A.end = tour1.forStructured[6];
+         A.start = tour1.forStructured[7];
+         __________________________________
+
+         B.start = tour2.forStructured[4];
+         B.end = tour2.forStructured[5];
+
+         or
+
+         B.end = tour2.forStructured[4];
+         B.start = tour2.forStructured[5];
+         __________________________________
+
+         C.start = tour4.forStructured[0];
+         C.end = tour4.forStructured[1];
+
+         or
+
+         C.end = tour4.forStructured[0];
+         C.start = tour4.forStructured[1];
+         __________________________________
+
+         D.start = tour2.forStructured[2];
+         D.end = tour2.forStructured[3];
+
+         or
+
+         D.end = tour2.forStructured[2];
+         D.start = tour2.forStructured[3];
+         __________________________________
+         */
+
+        /*
+         Deleting and simultaneously contracting.
+         */
+
+        if (tour1.getStructured().get(7).getNext() == tour1.getStructured().get(6)) {
```

Comment:
What do you think about only one if statement inside a for loop what goes through all the required tuples (7, 6, 2, 4) can be the first one for example? This will lead to shorter code and will probably be unrolled by the compiler.